src/App.jsx

```diff
@@ -14,45 +19,162 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function getPreparedGoods(goods, { sortField, reversed }) {
+  const preparedGoods = [...goods];
+  const preparedForReverse = [...preparedGoods];
+  const reversedGoods = preparedForReverse.reverse();
+
+  if (!reversed) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    reversedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          return good2.localeCompare(good1);
+
+        case SORT_FIELD_LENGTH:
+          return good2.length - good1.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reversed) {
+    return reversedGoods;
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reversed, setReversed] = useState(false);
```

Comment:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig